### PR TITLE
[ci] fix bazel log uploading conditions

### PR DESCRIPTION
**Pull request by can-anyscale**
Fix the conditions for NOT uploading bazel logs

Test:
- CI

### CODE REVIEW

File: ci/build/upload_build_info.sh [hunk 1]
```diff
@@ -6,9 +6,10 @@ set -ex
 readonly PIPELINE_POSTMERGE="0189e759-8c96-4302-b6b5-b4274406bf89"
 readonly PIPELINE_CIV1_BRANCH="0183465b-c6fb-479b-8577-4cfd743b545d"
 if [[
-  "${BUILDKITE_PIPELINE_ID:-}" != "${PIPELINE_POSTMERGE}" && 
-  "${BUILDKITE_PIPELINE_ID:-}" != "${PIPELINE_CIV1_BRANCH}" &&
-  "${BUILDKITE_BRANCH:-}" != "master"
+  (
+    "${BUILDKITE_PIPELINE_ID:-}" != "${PIPELINE_POSTMERGE}" && 
+    "${BUILDKITE_PIPELINE_ID:-}" != "${PIPELINE_CIV1_BRANCH}" 
+  ) || "${BUILDKITE_BRANCH:-}" != "master"
```
Comment (aslonnie) [inline ci/build/upload_build_info.sh:12]: maybe just split pipeline id from branch checking into 2 different `if` lines? so that the message is accurate?

**Review by aslonnie — APPROVED** [review]

**Comment by can-anyscale** [comment]
split the check into two if \o/

File: ci/build/upload_build_info.sh [hunk 2]
```diff
@@ -6,10 +6,14 @@ set -ex
...
-  "${BUILDKITE_BRANCH:-}" != "master"
+    "${BUILDKITE_PIPELINE_ID:-}" != "${PIPELINE_POSTMERGE}" && 
+    "${BUILDKITE_PIPELINE_ID:-}" != "${PIPELINE_CIV1_BRANCH}" 
 ]]; then
+  echo "Skip build info uploading on postmerge."
+  exit 0
+fi
+
+if [[ "${BUILDKITE_BRANCH:-}" != "master"]]; then
   echo "Skip build info uploading on master branch."
```
Comment (aslonnie) [inline ci/build/upload_build_info.sh:17]: nit: I think you mean skip uploading on "non-master" branch?

Comment (can-anyscale) [reply ci/build/upload_build_info.sh:17]: w00t

File: ci/build/upload_build_info.sh [hunk 3]
```diff
@@ -6,10 +6,14 @@ set -ex
 readonly PIPELINE_POSTMERGE="0189e759-8c96-4302-b6b5-b4274406bf89"
 readonly PIPELINE_CIV1_BRANCH="0183465b-c6fb-479b-8577-4cfd743b545d"
 if [[
-  "${BUILDKITE_PIPELINE_ID:-}" != "${PIPELINE_POSTMERGE}" && 
-  "${BUILDKITE_PIPELINE_ID:-}" != "${PIPELINE_CIV1_BRANCH}" &&
-  "${BUILDKITE_BRANCH:-}" != "master"
+    "${BUILDKITE_PIPELINE_ID:-}" != "${PIPELINE_POSTMERGE}" && 
+    "${BUILDKITE_PIPELINE_ID:-}" != "${PIPELINE_CIV1_BRANCH}" 
 ]]; then
+  echo "Skip build info uploading on postmerge."
```
Comment (aslonnie) [inline ci/build/upload_build_info.sh:12]: "Skip upload build info; we only upload on postmerge pipelines."

**Pull request merged**